spec/javascripts/components/story/expanded_story/expanded_story_labels_spec.js

```diff
@@ -3,39 +3,88 @@ import { shallow, mount } from 'enzyme';
 import ExpandedStoryLabels from 'components/story/ExpandedStory/ExpandedStoryLabels';
 
 describe('<ExpandedStoryLabels />', () => {
+  const defaultLabels = [
+    { id: 0, name: 'front' },
+    { id: 1, name: 'back' }
+  ];
+
   const defaultProps = () => ({
     onAddLabel: sinon.spy(),
     onRemoveLabel: sinon.spy(),
+    disabled: false,
     projectLabels: []
   });
+  
+  const setup = (propOverrides, labelOverrides) => {
+    const labels = labelOverrides || defaultLabels;
+    const story = { labels, _editing: { labels } };
+    const props = {
+      ...defaultProps(),
+      story,
+      ...propOverrides
+    };
 
-  it('renders component title', () => {
-    const labels = [
-      { id: 0, name: 'front' },
-      { id: 1, name: 'back' }
-    ];
+    const wrapper = shallow(<ExpandedStoryLabels {...props} />);
+    const reactTags = wrapper.find('ReactTags');
+    const { onRemoveLabel, onAddLabel } = props;
 
-    const story = { _editing: { labels } }
+    return { wrapper, reactTags, labels, onRemoveLabel, onAddLabel };
+  }
+
+  it('renders component title', () => {
+    const story = { _editing: { defaultLabels } }
     const wrapper = mount(
       <ExpandedStoryLabels {...defaultProps()} story={story} />
     );
-
+    
     expect(wrapper.text()).toContain(I18n.t('activerecord.attributes.story.labels'));
   });
 
+  describe('when component is disabled', () => {
+    it('does not allow new labels', () => {
+      const { wrapper } = setup({ disabled: true });
+
+      expect(wrapper.find('ReactTags').prop('allowNew')).toBe(false);
+    });
+
+    it('does not allow deleting labels', () => {
+      const { wrapper, onRemoveLabel } = setup({ disabled: true });
+
+      wrapper.instance().handleDelete(0);
+
+      expect(onRemoveLabel).not.toHaveBeenCalled();
+    });
+
+    describe('when story has no labels', () => {
+      it('renders nothing', () => {
+        const { wrapper } = setup({ disabled: true }, []);
+
+        expect(wrapper.type()).toBeFalsy();
```

Comment:
What is this actually testing?

Reply:
We need to test if the render method returns null, we can also use `expect(wrapper.htlm()).toBeNull()` which is more verbose.